lib/TPP/VNNIUtils.cpp

```diff
@@ -28,10 +31,17 @@ Optional<int64_t> getVNNIBlockingFactor(Type type) {
 }
 
 bool isBF16Type(Type type) {
-  if (!isa<ShapedType>(type))
+  auto elementType = getElementTypeOrSelf(type);
+  return elementType.isBF16();
+}
+
+bool isInVnniLayout(MemRefType memref) {
```

Comment:
Have you checked that any VNNI-related static local function to other files you create/use could be moved here?

Reply:
grepping a bit seems all clean.